### PR TITLE
Adds touch events to showcase events

### PR DESCRIPTION
**Pull request by amlcurran**
- Adds touch callbacks to showcase view
- Updates build tools
- Updates samples
- Adds a simple event listener to make overriding easier

wobble

cc @danybony ;)

### CODE REVIEW

File: sample/src/main/java/com/github/amlcurran/showcaseview/sample/EventsActivity.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        TextView eventLog = (TextView) findViewById(R.id.events_log);
+        Button customButton = (Button) getLayoutInflater().inflate(R.layout.view_custom_button, null);
+
+        new ShowcaseView.Builder(this)
+                .withMaterialShowcase()
+                .setStyle(R.style.CustomShowcaseTheme3)
+                .setTarget(new ViewTarget(R.id.imageView, this))
+                .setContentTitle("Events")
+                .setContentText("Listening to ShowcaseView events is easy!")
+                .setShowcaseEventListener(new MultiEventListener(new LogToTextListener(eventLog), new ShakeButtonListener(customButton)))
```
Comment (danybony) [inline sample/src/main/java/com/github/amlcurran/showcaseview/sample/EventsActivity.java]: Maybe extract the event listener in a dedicated local variable for better readability

File: sample/src/main/res/layout/activity_events.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright 2014 Alex Curran
```
Comment (danybony) [inline sample/src/main/res/layout/activity_events.xml]: 2015? :tongue: 
I never understood if you should use the creation date of the file or of the library, in this kind of comments..

File: library/src/main/java/com/github/amlcurran/showcaseview/OnShowcaseEventListener.java [hunk 3]
```diff
@@ -56,6 +58,12 @@ public void onShowcaseViewDidHide(ShowcaseView showcaseView) {
...
 
         }
+
+        @Override
+        public void onShowcaseViewTouchBlocked(MotionEvent motionEvent) {
+
+        }
     };
 
+    void onShowcaseViewTouchBlocked(MotionEvent motionEvent);
```
Comment (danybony) [inline library/src/main/java/com/github/amlcurran/showcaseview/OnShowcaseEventListener.java]: :-1: I'd move this before the `NONE` definition and add a javadoc description, as done for the other methods of this interface

Comment (amlcurran) [inline library/src/main/java/com/github/amlcurran/showcaseview/OnShowcaseEventListener.java]: Good catch :+1: https://github.com/amlcurran/ShowcaseView/commit/30da949150c1ce682ffc1a678581ff8e8bd2ee19

**Comment by danybony** [comment]
:100: LGTM! Ready to go as soon as the CI is green :green_heart:

**Comment by amlcurran** [comment]
@danybony it won't be, for some reason travis CI throws a lint error but local doesn't and I can't access the CI logs :( merging anyway #yolo

**Comment by danybony** [comment]
Lol! :+1:

**Pull request merged**